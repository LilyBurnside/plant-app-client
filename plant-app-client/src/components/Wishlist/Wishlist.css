*{
  font-family: 'Spartan';
}

.wishlist {
  display: flex;
  align-items: center;
  flex-direction: column;
}

img {
  width: 300px;
  margin-top: 50px;
}

.plant {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background-color: #DBFFB7;
  border-radius: 15px;
  margin-top: 50px;
  width: 100%;
  height: 800px;
}

.plant > p {
  margin-top: 20px;
}

.plant > a {
  text-decoration: none;
}

.quiz-restart {
  color: #494949 !important;
  text-decoration: none;
  background: #ffffff;
  padding: 20px;
  border-radius: 15px;
  border: 4px solid #494949 !important;
}