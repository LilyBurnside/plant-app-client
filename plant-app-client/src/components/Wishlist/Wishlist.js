import React from 'react'
import PlantsApiService from '../../services/plants-api-service'
import './Wishlist.css'

export default class Wishlist extends React.Component {

  state = { 
    error: null,
    wishlist: []
  }

  componentDidMount() {
    this.setState({ error: null })

    PlantsApiService.getWishlist()
      .then(this.setWishlist)
      .catch(res=> {
        this.setState({error: res.error})
      })
    console.log(this.state.wishlist)
  }

  setWishlist = wishlist => {
    this.setState({ wishlist })
  }

  handleQuizButton = e => {
    e.preventDefault()
    this.props.history.push('/quiz')
  }

  renderWishlist() {
    const { wishlist } = this.state.wishlist

    const wishlistArray = []
    if(!Array.isArray(wishlist) || !wishlist.length) {
      wishlistArray.push(<h2>Nothing here yet, take the quiz to add a plant!</h2>)
    } else {
      for(let i = 0; i < wishlist.length ; i++) {
        wishlistArray.push(
          <div className="plant">
            <img src={`https://${wishlist[i].photo}`} alt={wishlist[i].sci_name} />
            <h2>{!wishlist[i].cmn_name ? wishlist[i].sci_name : wishlist[i].cmn_name}</h2>
            <p>also known as: {wishlist[i].sci_name}</p>
            <p>try to find it from these online sources:</p>
            <a href="https://www.gabriellaplants.com/">Gabriella Plants</a>
            <a href="https://stevesleaves.com/">Steve's Leaves</a>
            <button className="delete">Delete plant</button>
          </div> 
        )
      }
    }
    return wishlistArray

  }

  render() {
    return(
      <div className="wishlist">
        <h1>Your Future Plants</h1>
        <p>Wishlisted plants live here until you find them!</p>
        {this.renderWishlist()}
        <button className="quiz-restart" onClick={this.handleQuizButton}>Match me with more plants</button>
      </div>
    )
  }
}
