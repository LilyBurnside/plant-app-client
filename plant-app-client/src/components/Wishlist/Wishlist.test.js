import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Wishlist from './Wishlist';

describe('Wishlist Component', () => {

  const Context = React.createContext({ wishlist: [{cmn_name: 'Pothos', sci_name: 'Epipremnum aureum', photo: 'https://plant-app-images.s3.us-east-2.amazonaws.com/epipremnum-aureum.jpg', light: 'medium', pet_safe: 'no', water: 'low', size: 'tabletop', care_level: 'beginner'}]})

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Wishlist/>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a message with no plants when empty', () => {
    const wrapper = shallow(<Wishlist />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders a wishlisted plant', () => {
    const wrapper = shallow(
      <Context.Provider>
        <Wishlist />
      </Context.Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  })

});