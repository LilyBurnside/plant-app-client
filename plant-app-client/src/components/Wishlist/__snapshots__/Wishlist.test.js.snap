// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wishlist Component renders a message with no plants when empty 1`] = `
<div
  className="wishlist"
>
  <h1>
    Your Future Plants
  </h1>
  <p>
    Wishlisted plants live here until you find them!
  </p>
  <div
    className="wishlist-plant-list"
  >
    <h2>
      Nothing here yet, take the quiz to add a plant!
    </h2>
  </div>
  <button
    className="quiz-restart"
    onClick={[Function]}
  >
    Match me with more plants
  </button>
</div>
`;

exports[`Wishlist Component renders a wishlisted plant 1`] = `<Wishlist />`;
